src/engine/Resource.ts

```diff
@@ -45,14 +45,16 @@ module ex {
        * @param responseType  The Content-Type to expect (e.g. `application/json`)
        * @param bustCache     Whether or not to cache-bust requests
        */
-      constructor(public path: string, public responseType: string, public bustCache: boolean = true) {}
+      constructor(public path: string, public responseType: string, public bustCache: boolean = true) {
+          super();
+      }
 
       /**
        * Returns true if the Resource is completely loaded and is ready
        * to be drawn.
        */
       public isLoaded(): boolean {
-         return !!this.data;
+         return this.data !== null;
```

Comment:
More than that, I loaded an empty text file so the value was `""` so this evaluated to false. Whoops! Comparing to null maybe even sometimes could cause an issue if null is a valid value to load...